{collectd,libcollectdclient}: fix build

philiptaron opened the pull request:
https://hydra.nixos.org/build/300269816

Three issues fixed:

1. Intermittent connection failures to https://collectd.org made the `src` derivation flaky, so switched to the upstream github repository.
2. No Bison or Flex listed as a dependency, so parser and scanner were not generated.
3. `src/virt.c` includes `ATTRIBUTE_UNUSED` but it's not defined anywhere.

Also switched to `finalAttrs` style and SRI hash format throughout, and reduced the scope of `with lib`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

bjornfor commented:
> Three issues fixed:

Please include this valuable info in the git commit message, which is the true nixpkgs changelog.

If I'm being nitpicky, I like to keep "fix build" and "modernize/clean up/churn" commits separate, but I won't go there now, as long as the commit explains all it does (like the PR).

philiptaron committed:
collectd: fix build and modernize derivation slightly

1. I experienced intermittent connection failures to <https://collectd.org> and I suspect it's hosted on hobbyist equipment.
   This made the `src` derivation flaky, so I switched to the upstream GitHub repository.
   It seems better to use the actual source instead of a prepared tarball from the release process anyway.

2. When building out of the source instead of the release tarball, we need to run Bison and Flex to generate the parser and lexer, and we need Perl for `pod2man`.

3. As the [build error on Hydra](https://hydra.nixos.org/build/300269816) notes, `src/virt.c` makes use of some token `ATTRIBUTE_UNUSED` but it's not defined anywhere. Define it.

4. Switch all hashes to use the SRI style.

5. Switch to `finalAttrs` fixpoint style for better overriding.

6. Reduce the scope of `with lib;` in the `meta` section.

Bit of a smörgåsbord but not too bad.

philiptaron committed:
libcollectdclient: fix build

I unified the `meta` field from `collectd` here too.

philiptaron commented:
Thanks for the reminder @bjornfor; I've done so. I also fixed up `libcollectdclient` as well.

philiptaron commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 418186`
Commit: `d72d4e7a8764a41bb95539c3fb987c9c4a7c9c28`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>libcollectdclient</li>
  </ul>
</details>

bjornfor commented:
Thanks!
